### DISCUSSION TITLE
Joystick on a Mac not working

### DISCUSSION THREAD

**Discussion by fedekrum** [winder/Universal-G-Code-Sender#2012] [Help]
Hi.
I am using de following:

Product Version: Universal Gcode Platform 20220810
Java: 13.0.1; OpenJDK 64-Bit Server VM 13.0.1+9
Runtime: OpenJDK Runtime Environment 13.0.1+9
System: Mac OS X version 10.15.7 running on x86_64; UTF-8; en_US (ugsplatform)

At https://gamepad-tester.com/ , I am able to see my USB gamepad with no problem

![Captura de Pantalla 2022-10-21 a la(s) 14 07 01](https://user-images.githubusercontent.com/4470683/197192155-1fec4fff-ef05-45f3-9524-a77e57fa8370.png)

But UGS is unable to detect the buttons when pressed at the setup window even after applying the "Active joystick" checkbox and pressing the "Apply" button.

![Captura de Pantalla 2022-10-21 a la(s) 14 09 27](https://user-images.githubusercontent.com/4470683/197192781-426a9fd1-6de1-4b31-a92d-6c9929a8023d.png)

I can confirm that I do use this gamepad on other applications on my Mac.

Any clue why it is not working?
Thanks

**Comment by breiler** [comment]
It looks like the controller is available in the game controller database: 
https://github.com/winder/Universal-G-Code-Sender/blob/master/ugs-platform/ugs-platform-plugin-joystick/src/main/resources/com/willwinder/ugs/nbp/joystick/gamecontrollerdb.txt#L456

I have no idea why it isn't detected. The logs might contain clues: `C:/Users/[your username>]/.ugsplatform/2.X/dev/var/log/messages.log`

**Reply by fedekrum** [reply]
I am on a Mac ...where is .ugsplatform folder ?

**Reply by breiler** [reply]
🤦 Of course here: `~/Library/Application Support/ugsplatform/2.X/dev/var/log/messages.log`

**Reply by fedekrum** [reply]
[log.zip](https://github.com/winder/Universal-G-Code-Sender/files/9839883/log.zip)

**Reply by breiler** [reply]
Nope it didn't give any clues, sorry.

**Reply by fedekrum** [reply]
The other thing, is that while trying to configure the joystick, after pressing "Apply", sometimes it closes the app suddenly. Not every time.

**Reply by joshua-caddell** [reply]
Was this issue ever resolved? I am having the same problem with UGS 2.0.13 on macOS 13.1 (Intel). UGS doesn’t recognize any input from my game controller.

**Reply by breiler** [reply]
@joshua-caddell what game controller do you have?

Note that if it is a XBox 360 compatible controller you usually need a special driver to make it work: https://github.com/360Controller/360Controller

**Reply by joshua-caddell** [reply]
@breiler It's a PS5 controller. They are natively supported by macos now and gamepad-tester does recognized does recognize it.

**Reply by breiler** [reply]
@joshua-caddell I updated the game controller database 2022-12-28 which should now include the PS5 Controller:
https://github.com/winder/Universal-G-Code-Sender/blob/master/ugs-platform/ugs-platform-plugin-joystick/src/main/resources/com/willwinder/ugs/nbp/joystick/gamecontrollerdb.txt#L919

It was changed after the 2.0.13 release which means that it should be available in the latest nightly builds.

**Reply by joshua-caddell** [reply]
@breiler thanks for the quick response. I'll install the nightly build later today and report back.
